lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -3812,7 +3808,11 @@ instance HasSeverityAnnotation TxSubmitLog where
 
 -- | Construct the default 'ChangeAddressGen s' for a given 's'.
 defaultChangeAddressGen
-    :: forall s. GenChange s
+    :: forall s (k :: Depth -> * -> *). (GenChange s, BoundedAddressLength k)
```

Comment:
IIRC, on GHC 9.2, using `*` will generate a warning.

But we can change it to `Type` (from `Data.Kind`):

```suggestion
    :: forall s (k :: Depth -> Type -> Type). ()
    => (GenChange s, BoundedAddressLength k)
```

Reply:
> IIRC, on GHC 9.2, using * will generate a warning.

Sure, I was using GHC 9.2 locally but actively ignored this as I wasn't certain 8.10 would be happy about it... But I think it will... have pushed a commit to check...